scrapy/exporters.py

```diff
@@ -86,10 +88,10 @@ def __init__(self, file, **kwargs):
 
     def export_item(self, item):
         itemdict = dict(self._get_serialized_fields(item))
-        self.file.write(self.encoder.encode(itemdict) + '\n')
+        self.file.write(to_bytes(self.encoder.encode(itemdict) + '\n'))
 
 
-class JsonItemExporter(JsonLinesItemExporter):
+class JsonItemExporter(BaseItemExporter):
```

Comment:
This is by design, `BaseItemExporter` implements the stuff common to all exporters (recursively export fields, fields_to_export and export_empty_fields settings).